azurerm/config.go

```diff
@@ -46,10 +46,11 @@ import (
 // ArmClient contains the handles to all the specific Azure Resource Manager
 // resource classes' respective clients.
 type ArmClient struct {
-	clientId       string
-	tenantId       string
-	subscriptionId string
-	environment    azure.Environment
+	clientId              string
+	tenantId              string
+	subscriptionId        string
+	usingServicePrincipal bool
```

Comment:
Nitpick, but `using*` to me sounds like a state that may be managed from somewhere outside, how do you feel about calling it `useServicePrinciple` instead?

Reply:
it will be managed from somewhere outside (it's determined in the Config class, and then cast to an `ArmConfig` once the authentication tokens have been obtained) - as such I think `using` makes more sense in this context than `use`? That said, I'm happy to change this - but I'm a little stumped for a better name ¯\_(ツ)_/¯

Reply:
(merging this for the moment - we can rename this later if needed)